proto/bpfd.proto

```diff
@@ -7,27 +7,71 @@ service Bpfd {
     rpc Unload (UnloadRequest) returns (UnloadResponse);
     rpc List (ListRequest) returns (ListResponse);
     rpc PullBytecode (PullBytecodeRequest) returns (PullBytecodeResponse);
+    rpc Get (GetRequest) returns ( GetResponse );
 }
 
+/* BytecodeImage represents an eBPF program that is packaged and contained within
+ * an OCI container image.
+ */
+
 message BytecodeImage {
     string url = 1;
     int32 image_pull_policy = 2;
-    string username = 3;
-    string password = 4;
+    optional string username = 3;
+    optional string password = 4;
 }
 
-message LoadRequestCommon {
+/* BytecodeLocation is either:
+ * - Parameters to pull an eBPF program stored in an OCI container image.
+ * - Local file path for an image.
+ */
+message BytecodeLocation { 
     oneof location {
-        BytecodeImage image = 1;
-        string file = 2;
-    } ;
-    string name = 3;
-    uint32 program_type = 4;
-    map<string,string> metadata = 5;
-    map<string, bytes> global_data = 6;
-    optional uint32 map_owner_id = 7;
+        BytecodeImage image = 2;
+        string file = 3;
+    }
+}
+
+/* KernelProgramInfo represents the state for a single ITS_BPF_NOT_EBPF program that is
+ * maintained internally by the Linux kernel within its bpf_prog_info
+ * structure.
+ */
+
+message KernelProgramInfo { 
+    uint32 id = 1;
+    /* Same a Name entered by user, but truncated to 16 byte by kernel */
+    string name = 2;
+    uint32 program_type = 3;
+    string loaded_at = 4;
+    string tag = 5;
+    bool gpl_compatible = 6;
+    repeated uint32 map_ids = 7;
+    uint32 btf_id = 8;
+    uint32 bytes_xlated = 9;
+    bool jited = 10;
+    uint32 bytes_jited = 11;
+    uint32 bytes_memlock = 12;
+    uint32 verified_insns = 13;
+}
+
+/* ProgramInfo represents the state for a single eBPF program that is maintained
+ * internally by bpfd. */
+
+message ProgramInfo {
+    /* Name entered by user */
+    string name = 1;
+    BytecodeLocation bytecode = 2;
+    AttachInfo attach = 3;
+    map<string, bytes> global_data = 4;
+    optional uint32 map_owner_id = 5;
+    optional string map_pin_path = 6;
```

Comment:
At least in our internal api map_pin_path isn't optional, and i'm not sure it should be here either.

Reply:
It's not entered by the user, but returned from bpfd. So needs to be optional.

Reply:
I'm fine changing it here in the external API to optional, but it is optional in the internal API (command.rs: ProgramData).

Reply:
Done